### PR TITLE
ctl: improve error handling

### LINKED ISSUE #77: ctl: make `list-processes` not case sensitive
`list-processes` currently requires the peer flag to be all lowercase. 

The `FromStr` trait for `MaybeAllPeerId` should make the input string all lowercase before working with it.

### LINKED ISSUE #131: ctl: improve error handling
currently ctl panics on any error. Every subcommand's `run` function should return a result so we can use the `?` operator to propagate the error to main and handle it there. We should use `anyhow` for human readable errors and yacexits for exit codes.

### PR DESCRIPTION
**Pull request by airidaceae**
closes #131 
closes #77 

This pr still needs a lot of work before it can be merged. I would also like to have someone look over my use of error codes and make sure I used the right ones.

### CODE REVIEW

File: crates/hearth-ctl/src/list_processes.rs [hunk 1]
```diff
@@ -72,37 +74,52 @@ impl ListProcesses {
                         is_first = false;
                     }
                     ListProcesses::display_peer(
-                        daemon.peer_provider.find_peer(id).await.unwrap(),
+                        daemon
+                            .peer_provider
+                            .find_peer(id)
+                            .await
+                            .to_command_error("finding peer", EX_PROTOCOL)?,
```
Comment (marceline-cramer) [inline crates/hearth-ctl/src/list_processes.rs:81]: Make this peer provider function a crate-level helper function that fetches a peer from a daemon and returns this same error message. Replace all peer location operations with this function.

File: crates/hearth-ctl/src/spawn_wasm.rs [hunk 2]
```diff
@@ -34,31 +37,42 @@ pub struct SpawnWasm {
...
             .follow_service_list()
             .await
-            .unwrap()
+            .to_command_error("following service list", EX_PROTOCOL)?
             .take_initial()
-            .unwrap()
+            .to_command_error("getting service list", EX_PROTOCOL)?
             .get("hearth.cognito.WasmProcessSpawner")
-            .expect("Peer is not running WebAssembly")
+            .to_command_error("retreiving WasmProcessSpawner", EX_UNAVAILABLE)?
```
Comment (marceline-cramer) [inline crates/hearth-ctl/src/spawn_wasm.rs:59]: retreiving -> retrieving

File: crates/hearth-ctl/src/main.rs [hunk 3]
```diff
@@ -72,27 +109,33 @@ pub enum Commands {
...
+        }
+    }
 }
 
-async fn get_daemon() -> DaemonOffer {
+async fn get_daemon() -> Result<DaemonOffer, CommandError> {
     hearth_ipc::connect()
         .await
-        .expect("Failed to connect to Hearth daemon")
+        .to_command_error("connecting to hearth daemon", yacexits::EX_UNAVAILABLE)
```
Comment (marceline-cramer) [inline crates/hearth-ctl/src/main.rs:138]: capitalize the "h" in "Hearth"

**Review by marceline-cramer — CHANGES_REQUESTED** [review]
Looks generally beautiful. A couple things:
- replace all `Result<(), CommandError>` with a generic type at the crate level: `type CommandResult<T> = Result<T, CommandError>`, then update all instances of that type to match.
- compact as much redundant code as possible into helper functions on the crate level. i want as little duplicated code as possible.

**Comment by airidaceae** [comment]
closes #77

**Pull request merged**